release/lib/git-helpers.ts

```diff
@@ -0,0 +1,310 @@
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import { promisify } from 'util';
+
+import * as req from 'request';
+import { Tag, Commit, Package, Author, GitHubAuth } from '../@types/custom';
+import { debug, execa, REPOSITORY_SLUG } from './utils';
+
+const request = promisify(req) as (options: req.OptionsWithUrl) => Promise<req.Response>;
+
+const GITHUB = {
+    password: '',
+    token: '',
+    tokenID: '',
+    userName: ''
+};
+
+const USER_AGENT = 'Nellie The Narwhal';
+
+const authors: Map<string, Promise<Author | null>> = new Map();
+const commitAuthors: Map<string, Promise<Author | null>> = new Map();
+/**
+ * Returns the GitHub user information associated with a give commit.
+ * @param commitSHA The sha of the commit to get the information from.
+ */
+const getCommitAuthorInfo = (commitSHA: string): Promise<Author | null> => {
```

Comment:
I originally had it with `async/await` but because the info for the commits is run in parallel I was be querying the same information multiple times and reach the GitHub limit for requests and everything crashed. Now that everything is stored as promises we only request the bare minimum.